test/automated-browser-tests.js

```diff
@@ -0,0 +1,248 @@
+          return driver.executeScript('return ((typeof window.testsuite !== \'undefined\') && window.testsuite.testResults !== \'undefined\');');
+        });
+      })
+      .then(() => {
+        // This simply retrieves the test results from the inbrowser mocha tests
+        return driver.executeScript('return window.testsuite.testResults;');
+      })
+      .then(testResults => {
+        // Resolve the outer promise to get out of the webdriver promise chain
+        resolve(testResults);
+      })
+      .thenCatch(reject);
+    })
+    .then(testResults => {
+      if (testResults.failed.length > 0) {
+        const failedTests = testResults.failed;
+        let errorMessage = 'Issues in ' + browserName + '.\n\n' + browserName + ' had ' + testResults.failed.length + ' test failures.\n';
+        errorMessage += '------------------------------------------------\n';
+        errorMessage += failedTests.map((failedTest, i) => {
+          return `[Failed Test ${i + 1}]\n    ${failedTest.title}\n`;
+        }).join('\n');
+        errorMessage += '------------------------------------------------\n';
+        throw new Error(errorMessage);
+      }
+    });
+  };
+
+  const queueUnitTest = (browserName, browserPath, seleniumBrowserID, options) => {
+    if (!browserPath) {
+      console.warn(`${browserName} path wasn\'t found so skipping`);
+      return;
+    }
+
+    it(`should pass all tests in ${browserName}`, () => {
+      globalDriverReference = new webdriver
+        .Builder()
+        .forBrowser(seleniumBrowserID)
+        .setChromeOptions(options)
+        .setFirefoxOptions(options)
+        .build();
+
+      return performTests(browserName, globalDriverReference);
+    });
+  };
+
+  function configureBrowserTests() {
+    // Chrome Stable
+    try {
+      const CHROME_PATH = which.sync('google-chrome');
+      const chromeStableOpts = new chromeOptions.Options();
+      chromeStableOpts.setChromeBinaryPath(CHROME_PATH);
+      queueUnitTest('Chrome Stable', CHROME_PATH, 'chrome', chromeStableOpts);
+    } catch (err) {
+      console.error('Unable to find Chrome Stable.');
+    }
+
+
+    // Chrome Beta
+    try {
+      const CHROME_BETA_PATH = which.sync('google-chrome-beta');
+      const chromeBetaOpts = new chromeOptions.Options();
+      chromeBetaOpts.setChromeBinaryPath(CHROME_BETA_PATH);
+      queueUnitTest('Chrome Beta', CHROME_BETA_PATH, 'chrome', chromeBetaOpts);
+    } catch (err) {
+      console.error('Unable to find Chrome Beta.');
+    }
+
+    // Firefox Default Install
+    try {
+      const FIREFOX_PATH = which.sync('firefox');
+      const ffStableOpts = new firefoxOptions.Options();
+      ffStableOpts.setBinary(FIREFOX_PATH);
+      queueUnitTest('Firefox Stable', FIREFOX_PATH, 'firefox', ffStableOpts);
+    } catch (err) {
+      console.error('Unable to find default Firefox.');
+    }
+
+
+    // Firefox Beta in specific path on Travis
+    if (process.env.TRAVIS) {
```

Comment:
Maybe turn that `if` into something like `if (process.env.TRAVIS || process.env.FF_BETA_PATH)`? So at least you can run FF Beta without diving into the test code. WDYT?